packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -214,7 +220,10 @@ export abstract class RetrieveExecutor<T> extends DeployRetrieveExecutor<T> {
 
     this.setupCancellation(operation, token);
 
-    return operation.pollStatus();
+    const result: RetrieveResult = await operation.pollStatus();
+    await sourceTracking.updateTrackingFromRetrieve(result);
```

Comment:
Had to refactor this section a little bit.  The call to updateTrackingFromRetrieve() is needed to update source-tracking after a new file that is created on the server is retrieved.